### ISSUE TITLE
Title bar in dark color

### ISSUE THREAD

**Issue by ANF-Studios** [TheCherno/Hazel#420]
#### Is your feature request related to a problem? Please describe
The title bar of the engine UI is in light mode

#### Describe the solution you'd like
I don't know if this is possible, but it would be nice if the title bar would also be in dark mode since the in-engine UI is in dark mode.

#### Describe alternatives you've considered
I recall that it's possible to make the window's title bar frameless and make a custom one but I'm not sure if that'll be a great idea.

#### Additional context
A screenshot of the editor from [this video](https://www.youtube.com/watch?v=GKERbQ7Sqyg&list=PLlrATfBNZ98dC-V-N3m0Go4deliWHPFwT&index=98):
![image](https://user-images.githubusercontent.com/68814933/107635190-8ce57000-6c8c-11eb-965a-3485ad62daf7.png)

**Comment by LovelySanta** [comment]
In the past we tried implementing borderless mode, which hides the title bar outside the rendered area of the screen. This is more a setting controlled by windows itself, for me, windows is set in dark mode, resulting in a black bar on top as well.

Basic applications such as Word and Excell have their own custom title bar, so I we could try and implement our own variant, by hiding the default one, or altering it as we see fit (if possible).

**Comment by YusufSuleman** [comment]
I remember hearing about a fix for this. You need to tell Windows this is in Dark Mode and it will do it. Linux GNOME has Dark Mode Globally. I don't know anything about Mac.

**Comment by LovelySanta** [comment]
https://github.com/glfw/glfw/issues/1516 is the closest i've found while researching this issue...

**Comment by AnantTiwari-Naman** [comment]
I am pretty sure this is a windows specific thing rather than a Hazel / GLFW thing. I remember changing some setting in windows, which made the title bar of all windows follow my main accent color, I don't get the white title bar on Hazel.

**Comment by ANF-Studios** [comment]
> I am pretty sure this is a windows specific thing rather than a Hazel / GLFW thing. I remember changing some setting in windows, which made the title bar of all windows follow my main accent color, I don't get the white title bar on Hazel.
Well yes, that is absolutely true, but I am pretty sure there is some way to do it with code. Like UWP does it.

Oh and, you can do that (on Windows) by applying the accent color of your pc to title bars as well:
![image](https://user-images.githubusercontent.com/68814933/113479324-77a2eb80-945c-11eb-8a32-2eca3726572f.png)
The only downside is that, for that to happen, it won't be very convenient, and secondly; not everyone prefers to change their color to dark on other applications, globally.

TL;DR: There are downsides to that, as a developer, products should more be focused on handling for the user rather than by the user even if it's a game engine.


Oh and, I think *maybe* we can use [DwmSetWindowAttribute](https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmsetwindowattribute) on Windows to do that. *I guess* it might be possible. Not sure though.

**Comment by AnantTiwari-Naman** [comment]
Sure, we can make the window borderless, and manually add the window controls and stuff by our own.

**Comment by LovelySanta** [comment]
> Sure, we can make the window borderless, and manually add the window controls and stuff by our own.

isn't borderless always fullscreen?

**Comment by AnantTiwari-Naman** [comment]
There's a borderless-windowed mode, which (from what I understand) is the "normal" borderless mode, many games make their windows borderless and resize them so it takes the whole screen, but just refer to it as borderless mode

**Comment by ANF-Studios** [comment]
> > Sure, we can make the window borderless, and manually add the window controls and stuff by our own.
> 
> isn't borderless always fullscreen?

Yea, it's possible. I did [find](https://github.com/rossy/borderless-window) some [repositories](https://github.com/ands/borderless-window-opengl) with examples of borderless windows, not sure if they're full screen though, probably not. But yes, it is possible, now.. I don't write a lot of C++ code, I've been writing a JavaScript desktop application which uses a borderless window and a "virtual" title bar, the way it handles it is that it takes away the frame, and handles the UI and all the details itself. This is how it looks:
![image](https://user-images.githubusercontent.com/68814933/113786624-7f39ed00-9707-11eb-84a7-1dc6a28cbc7b.png)
This window is very different from a normal one and I am sure you can tell the difference.

However, again; I do think dark mode can *also* be enabled by using Dwm (on Windows).

**Comment by AnantTiwari-Naman** [comment]
Yep, that's what I was thinking about
